core/sr-primitives/src/generic/unchecked_extrinsic.rs

```diff
@@ -71,38 +78,47 @@ where
 		UncheckedExtrinsic {
 			signature: None,
 			function,
+			tip: Tip::None,
 		}
 	}
 }
 
-impl<Address, Index, Signature, Call, AccountId, Context> traits::Checkable<Context>
-	for UncheckedExtrinsic<Address, Index, Call, Signature>
+impl<Address, Index, Signature, Call, AccountId, Context, Balance> traits::Checkable<Context>
+	for UncheckedExtrinsic<Address, Index, Call, Signature, Balance>
 where
 	Address: Member + MaybeDisplay + Codec,
 	Index: Member + MaybeDisplay + SimpleArithmetic + Codec,
 	Call: Encode + Member,
 	Signature: Member + traits::Verify<Signer=AccountId> + Codec,
 	AccountId: Member + MaybeDisplay,
 	Context: Lookup<Source=Address, Target=AccountId>,
+	Balance: Member + Encode,
 {
-	type Checked = CheckedExtrinsic<AccountId, Index, Call>;
+	type Checked = CheckedExtrinsic<AccountId, Index, Call, Balance>;
 
 	fn check(self, context: &Context) -> Result<Self::Checked, &'static str> {
 		Ok(match self.signature {
 			Some(SignatureContent{signed, signature, index}) => {
-				let payload = (index, self.function);
+				let payload = (index, self.function, self.tip);
 				let signed = context.lookup(signed)?;
 				if !crate::verify_encoded_lazy(&signature, &payload, &signed) {
 					return Err(crate::BAD_SIGNATURE)
 				}
 				CheckedExtrinsic {
 					signed: Some((signed, payload.0)),
 					function: payload.1,
+					tip: payload.2,
 				}
 			}
-			None => CheckedExtrinsic {
-				signed: None,
-				function: self.function,
+			None => {
+				if self.tip == Tip::None {
```

Comment:
```suggestion
				if self.tip != Tip::None {
```

We should have a test for it I suppose

Reply:
There is: 

```
#[test]
	#[should_panic]
	fn unsigned_cannot_have_tip() {
		let ux = UncheckedMortalCompactExtrinsic { signature: None, tip: Tip::Sender(100), function: vec![0u8;0]};
		let _ = <Ex as Checkable<TestContext>>::check(ux, &TestContext).unwrap();
	}
``` 

but I placed it in the `mortal` versions since I assume the `unchecked_extrinsic` is not really maintained yet. I asked it in a chat room; if we are maintaining the raw `unchecked_extrinsic` as well then I should add a bunch of tests to it.